File: skellig-test-step-processing-cassandra/src/main/java/org/skellig/teststep/processor/cassandra/CassandraRequestExecutorFactory.java
package org.skellig.teststep.processor.cassandra;

import com.datastax.driver.core.Session;
import org.skellig.teststep.processor.db.DatabaseRequestExecutorFactory;

class CassandraRequestExecutorFactory extends DatabaseRequestExecutorFactory {

    CassandraRequestExecutorFactory(Session session) {
        super(new CassandraSelectRequestExecutor(session), new CassandraInsertRequestExecutor(session));
    }

}

File: skellig-test-step-processing-cassandra/src/main/kotlin/org/skellig/teststep/processor/cassandra/CassandraRequestExecutorFactory.kt
package org.skellig.teststep.processor.cassandra

import com.datastax.oss.driver.api.core.CqlSession
import org.skellig.teststep.processor.db.DatabaseRequestExecutorFactory

internal class CassandraRequestExecutorFactory(session: CqlSession)
    : DatabaseRequestExecutorFactory(
        CassandraSelectRequestExecutor(session),
        CassandraInsertRequestExecutor(session),
        CassandraUpdateRequestExecutor(session,
                CassandraInsertRequestExecutor(session),
                CassandraSelectRequestExecutor(session)))